compiler/rustc_infer/src/infer/error_reporting/suggest.rs

```diff
@@ -351,6 +351,73 @@ impl<'tcx> TypeErrCtxt<'_, 'tcx> {
         }
     }
 
+    pub(super) fn suggest_function_pointers(
+        &self,
+        exp_found: &ty::error::ExpectedFound<Ty<'tcx>>,
+        diag: &mut Diagnostic,
+    ) {
+        let ty::error::ExpectedFound { expected, found } = exp_found;
+        if expected.peel_refs().is_fn() && found.peel_refs().is_fn() {
```

Comment:
tiny nit: can invert the logic here and do an early `return` to save a layer of indentation :smile: